sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -436,7 +436,8 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
           Try(fs.listStatus(qualified)).getOrElse(Array.empty)
         }.filterNot { status =>
           val name = status.getPath.getName
-          name.toLowerCase == "_temporary" || name.startsWith(".")
+          // Is it safe to replace "_temporary" to "_"?
```

Comment:
No. We need to preserve files like Parquet summary files (`_metadata` and `_common_metadata`).